feature/permissions/src/main/java/com/google/jetpackcamera/permissions/ui/PermissionsScreenComponents.kt

```diff
@@ -37,41 +37,67 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.graphics.painter.Painter
-import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import com.google.accompanist.permissions.ExperimentalPermissionsApi
 import com.google.accompanist.permissions.PermissionState
-import com.google.jetpackcamera.R
+import com.google.accompanist.permissions.shouldShowRationale
+import com.google.jetpackcamera.permissions.PermissionEnum
+import com.google.jetpackcamera.permissions.R
 
+/**
+ * Template for a single page for the permissions Screen
+ *
+ * @param permissionEnum a [PermissionEnum] representing the target permission
+ * @param permissionState a [PermissionState] of the target permission
+ */
 @OptIn(ExperimentalPermissionsApi::class)
 @Composable
-fun CameraPermission(cameraPermissionState: PermissionState, modifier: Modifier = Modifier) {
+fun PermissionTemplate(
+    modifier: Modifier = Modifier,
+    permissionEnum: PermissionEnum,
+    permissionState: PermissionState,
+    onRequestPermission: () -> Unit,
+    onSkipPermission: (() -> Unit)? = null,
+    onOpenAppSettings: () -> Unit
+) {
     PermissionTemplate(
         modifier = modifier,
-        permissionState = cameraPermissionState,
-        painter = painterResource(id = R.drawable.photo_camera),
-        iconAccessibilityText = stringResource(id = R.string.camera_permission_accessibility_text),
-        title = stringResource(id = R.string.camera_permission_screen_title),
-        bodyText = stringResource(id = R.string.camera_permission_required_rationale),
-        requestButtonText = stringResource(R.string.request_permission)
+        onRequestPermission = {
+            // if declined by user, must navigate to system app settings to enable permission
+            // todo(b/337338326): open a dialog that tells user they must go to device's app settings to enable permissions
+            if (permissionState.status.shouldShowRationale) {
```

Comment:
Does this take into account that the developer might have purposefully denied a permission? When I deny the audio permission, I'm still being asked to go to the app settings.

Reply:
this is only used when a permission is flagged as required (which is just camera permission). If a user declines a permission flagged as optional (microphone), the screen closes after declining and it will not be prompted again when the app is reopened.
> When I deny the audio permission, I'm still being asked to go to the app settings.

I didn't have this issue on my devices, I'll check again on some other devices and to make sure the logic is right.

Reply:
It's still happening on my S23 Ultra. Except now it crashes on line 87 when I deny the permission with a `NullPointerException` because the audio permission doesn't have a rationale body text.

Reply:
After checking out the latest code, it no longer crashes. I was able to get into a state where the spinner was going forever saying the camera was loading, but that may be unrelated to this change. We can tackle that as a separate bug.